src/2-2_Compute-metrics_doublets.r

```diff
@@ -0,0 +1,45 @@
+################################################
+## 2-2_Compute-metrics_scrublet.r
+##  - Compute doublet scores per sample
+##  - Save an txt file with doublet scores per sample [doublet-scores-sampleName]
+################################################
+
+suppressWarnings(library(Seurat))
+suppressWarnings(library(scDblFinder))
+
+message("reloading old matrices...")
+scdata_list <- readRDS("/output/pre-doublet-scdata_list.rds")
+
+message("Loading configuration...")
+config <- RJSONIO::fromJSON("/input/meta.json")
+
+# Check which samples have been selected. Otherwiser we are going to use all of them. 
+if (length(config$samples)>0){
+    samples <- config$samples
+}else{
+    samples <- names(scdata_list)
+}
+
+scdata_list <- scdata_list[samples]
+
+# compute_doublet_scores function
+#' @description Save the result of doublets scores per sample. 
+#' @param scdata Raw sparse matrix with the counts for one sample.
+#' @param sample_name Name of the sample that we are preparing.
+#'
+#' @return 
+compute_doublet_scores <- function(scdata, sample_name, min.features = 10) {
+  
+    message("Sample --> ", sample_name, "...")
+    scdata_DS <- scDblFinder(scdata[, Matrix::colSums(scdata>0)>=min.features])
+    df_doublet_scores <- data.frame(Barcodes=rownames(scdata_DS@colData), doublet_scores=scdata_DS@colData$scDblFinder.score)
```

Comment:
```suggestion
    df_doublet_scores <- data.frame(Barcodes=rownames(scdata_DS@colData), doublet_scores=scdata_DS@colData$scDblFinder.score,
    doublet_class = scdata_DS@colData$scDblFinder.class)
```

We should store what `scDblFinder` classifies as a doublet as it is part of the algorithm (not just the score) and should be used to set the default threshold. [Here](https://bioconductor.org/packages/release/bioc/vignettes/scDblFinder/inst/doc/2_scDblFinder.html#thresholding-and-local-calibration) is the relevant part of the vignette that describes thresholding on the doublet score. If you agree, please make sure it is also being stored in the final Seurat object as well.

Reply:
I agree. I'll change the code to add it also in the seurat object